### ISSUE TITLE
New set of rules regarding ES6 modules

### ISSUE THREAD

**Issue by dispix** [eslint/eslint#7705]
**Please describe what the rule should do:**
As some people in the Javascript community start to express the benefits of using only `export default`, I suggest a new set of rules under something like `"es6-modules"`:

- `default-only`: disallow the usage of named exports
- `end-of-file-export`: only allow the usage of `export` at the end of the file

**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

This would be valid:
```js
const api = {
  add: (a, b) => a + b,
  substract: (a, b) => a - b
}

export default api
```

This would be invalid if enforcing `end-of-file-export`:
```js
export default const api = {
  add: (a, b) => a + b,
  substract: (a, b) => a - b
}
```

This would be invalid if enforcing `default-only`:
```js
const add = (a, b) => a + b
const substract = (a, b) => a - b

export add
export substract
```

This would also be invalid if enforcing `default-only`:
```js
export const add = (a, b) => a + b
export const substract = (a, b) => a - b
```

**Why should this rule be included in ESLint (instead of a plugin)?**

I don't really know how to justify that really, I feel like ES6 modules are now widely adopted and this shouldn't be part of a module.

Sorry if this is something that has already been discussed, I looked around but found nothing similar.

**Comment by gyandeeps** [comment]
I would recommend using [eslint-plugin-import](https://www.npmjs.com/package/eslint-plugin-import) for ES6 modules since it has lot of things already covered. Also if what you are looking for is not in that plugin, I would open an issue their.

**Comment by vitorbal** [comment]
Agreed with @gyandeeps. Looks like eslint-plugin-import is what you are looking for.

**Comment by dispix** [comment]
That doesn't exist in the plug-in but it might be a better place to put these rules indeed.

**Comment by mysticatea** [comment]
Though, I think good if rules of ES module syntaxes are in core because it's in the language spec.
On the other hand, I think that rules of ES module's lookup way fit to plugins because it's not in the language spec.

"To put imports at the head of files" and "to put exports at the last of files" are reasonable to me.

I'm not sure about `default-only`.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
